collections/src/main/scala/strawman/collection/immutable/Map.scala

```diff
@@ -128,27 +128,19 @@ trait MapOps[K, +V, +CC[X, +Y] <: MapOps[X, Y, CC, _], +C <: MapOps[K, V, CC, C]
   */
 object Map extends MapFactory[Map] {
 
-  final class WithDefault[K, +V](underlying: Map[K, V], d: K => V) extends Map[K, V] {
-    // These factory methods will lose the default value
-    def iterableFactory = underlying.iterableFactory
-    def mapFactory: MapFactory[Map] = underlying.mapFactory
-    protected[this] def mapFromIterable[K2, V2](it: collection.Iterable[(K2, V2)]): Map[K2,V2] = mapFactory.from(it)
-
-    // Specific building will keep the default but may lose the precise underlying type because our own V can be
-    // a supertype of the underlying collection's V so we cannot rebuild with potentially new values that are not
-    // valid for the underlying collection.
-    protected[this] def fromSpecificIterable(coll: collection.Iterable[(K, V)]): Map[K,V] = new WithDefault[K, V](mapFactory.from(coll), d)
-    protected[this] def newSpecificBuilder(): mutable.Builder[(K, V), Map[K,V]] =
-      mapFactory.newBuilder[K, V]().mapResult(new WithDefault[K, V](_, d))
-    override def size = underlying.size
-    def get(key: K) = underlying.get(key)
-    def iterator() = underlying.iterator()
-    override def default(key: K): V = d(key)
-    override def empty = new WithDefault(underlying.empty, d)
-    override def updated[V1 >: V](key: K, value: V1): WithDefault[K, V1] = new WithDefault[K, V1](underlying.updated[V1](key, value), d)
-    override def remove (key: K): WithDefault[K, V] = new WithDefault(underlying - key, d)
-    override def withDefault[V1 >: V](d: K => V1): immutable.Map[K, V1] = new WithDefault[K, V1](underlying, d)
-    override def withDefaultValue[V1 >: V](d: V1): immutable.Map[K, V1] = new WithDefault[K, V1](underlying, x => d)
+  final class WithDefaultMap[K, +V](val underlying: Map[K, V], val defaultValue: K => V) extends Map[K, V] with WithDefaultImmutableMapOps[K, V, Map[K, V]] {
+    override def remove(key: K): Map[K, V] = new WithDefaultMap[K, V](underlying - key, defaultValue)
```

Comment:
Could you please remove the `override` keyword`?

Reply:
Sorry again. Will do that. I have got Intellij template which adds them, will disable it for future commits to strawman-collection. :)